Backpack/Calendar/Classes/BPKCalendarCell.m

```diff
@@ -130,6 +130,7 @@ - (void)layoutSubviews {
         }
         break;
 
+    case SelectionTypeTrailing:
```

Comment:
not this line's fault..
But adding this here seems random, it's clearer and clearer each line I read that this needs refactoring, big time.

Reply:
maybe like having  different renderers for different kind of cells? or maybe even registering different cells for different kind of 'selection types'?

if going with renderer:
```swift
-(void) layoutSubviews {
   if wholeMonthSelection {
       wholeMonthCellRenderer.layoutCell(self)
       return
   }
   defaultCellRenderer.layourCell(self)
}
```
or if going with differnent cells
```swift
// on the BPKCalendar...
calendar.registerCell(WholeMonthCell.self, "WholeMonthSelection")
calendar.registerCell(DefaultCalendarCell.self, "DefaultCell")
//....
-(Cell)dequeueCalendarCell......() {
   if wholeMonthSelection {
       return dequeueCellWithIdentifier("WholeMonthSelection")
   }
   return dequeueCellWithIdentifier("DefaultCell")
}
```